### PR TITLE
[10.0][MIG] port purchase_analytic to V10.0

### PR DESCRIPTION
**Pull request by Cedric-Pigeon**
Migrate 8.0 PR to V10.0: https://github.com/OCA/account-analytic/pull/55

### CODE REVIEW

**Comment by Cedric-Pigeon** [comment]
@gurneyalex Could you please add the need review label ?

**Review by JonathanNEMRY — APPROVED** [review]

**Comment by JonathanNEMRY** [comment]
tested and backported into v9
@gurneyalex  I think it is more a `need review` than an `in progress` label :sweat_smile:

File: purchase_analytic/models/purchase.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        string='Contract / Analytic',
+        help='Use to store the value of project_id if there is no lines')
+    project_id = fields.Many2one(
+        compute='_compute_project_id',
+        inverse='_inverse_project_id',
+        comodel_name='account.analytic.account',
+        string='Contract / Analytic', readonly=True,
+        states={'draft': [('readonly', False)]},
+        store=True,
+        help="The analytic account related to a sales order.")
```
Comment (AaronHForgeFlow) [inline purchase_analytic/models/purchase.py:22]: "The analytic account related to a **purchase** order."

**Review by AaronHForgeFlow — APPROVED** [review]
:+1:  Code review

**Comment by aheficent** [comment]
@Cedric-Pigeon  Could you add some tests?

**Comment by feketemihai** [comment]
@Cedric-Pigeon I think first we should have the module merged in 8, and after to migrate it, check also my comment on #55 .

File: purchase_analytic/models/purchase.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+# © 2016  Laetitia Gangloff, Acsone SA/NV (http://www.acsone.eu)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    project_id2 = fields.Many2one(
```
Comment (MiquelRForgeFlow) [inline purchase_analytic/models/purchase.py:11]: maybe it would be better to call it `manual_project_id`?

**Review by MiquelRForgeFlow — APPROVED** [review]

**Comment by MiquelRForgeFlow** [comment]
@feketemihai I don't think we should have to wait the merging of v8. @Cedric-Pigeon, could you attend comments?

**Comment by AaronHForgeFlow** [comment]
@Cedric-Pigeon  Is is possible to attend comments here and move forward?

File: purchase_analytic/models/purchase.py (hunk 2 above)
Comment (AaronHForgeFlow) [reply purchase_analytic/models/purchase.py:11]: From my side it is fine to keep the field name like this.

**Comment by AaronHForgeFlow** [comment]
@Cedric-Pigeon can you rebase to check if ci is still green?

**Comment by Cedric-Pigeon** [comment]
@aheficent finally done...

**Comment by AaronHForgeFlow** [comment]
@pedrobaeza Can the issue with the CLA here block the Pull Request?

**Comment by pedrobaeza** [comment]
No, that one can be ignored as former member of Acsone. We should provide somehow a way to bypass it, but it's not yet implemented

**Pull request merged**